src/menuFactory.js

```diff
@@ -186,7 +186,7 @@ export default (styles) => {
 
     componentWillReceiveProps(nextProps) {
       // Allow open state to be controlled by props.
-      if (nextProps.isOpen !== this.state.isOpen) {
+      if (nextProps.isOpen !== this.props.isOpen && nextProps.isOpen !== this.state.isOpen) {
```

Comment:
I think we only care about state here (comparing with `this.props.isOpen` was the source of the bug), so this line could just become:
```
if (nextProps.isOpen !== this.state.isOpen) {
```